iroha/src/genesis.rs

```diff
@@ -1,14 +1,35 @@
 //! This module contains execution Genesis Block logic, and GenesisBlock definition.
-use crate::{init::config::InitConfiguration, tx::Accept, tx::AcceptedTransaction, Identifiable};
+use self::config::GenesisConfiguration;
+use crate::{
+    sumeragi::{InitializedNetworkTopology, Sumeragi},
+    torii::uri,
+    tx::Accept,
+    tx::AcceptedTransaction,
+    Identifiable,
+};
+use async_std::{
+    sync::{Arc, RwLock},
+    task,
+};
+use futures::future;
 use iroha_crypto::KeyPair;
 use iroha_data_model::{account::Account, isi::Instruction, prelude::*};
+use iroha_network::{Network, Request, Response};
 use serde::Deserialize;
-use std::{fmt::Debug, fs::File, io::BufReader, path::Path};
+use std::{fmt::Debug, fs::File, io::BufReader, path::Path, time::Duration};
+
+/// Time to live for genesis transactions.
+const GENESIS_TRANSACTIONS_TTL_MS: u64 = 100_000;
+
 /// GenesisBlock contains transactions for inisialize settings.
 #[derive(Clone, Debug)]
 pub struct GenesisBlock {
     /// transactions from GenesisBlock, any transaction is accepted
     pub transactions: Vec<AcceptedTransaction>,
+    /// Number of attempts to connect to peers, while waiting for them to submit genesis.
+    pub wait_for_peers_retry_count: u64,
+    /// Period in milliseconds in which to retry connecting to peers, while waiting for them to submit genesis.
+    pub wait_for_peers_retry_period_ms: u64,
```

Comment:
I admit it does seem a bit awkward, but it might be okay as the GenesisBlock is a special case. Maybe move the peers retry counts to something like a GenesisNetwork

Reply:
Yes, there might be some confusion here, as genesis block is not submitted by itself but instead just manages a process of submitting an ordinary block with the initial transaction. I will try the suggestion of introducing `GenesisNetwork` for better naming.